pkg/app/piped/analysisprovider/metrics/datadog/datadog.go

```diff
@@ -75,18 +68,27 @@ func NewProvider(apiKey, applicationKey string, queriedTimePeriod time.Duration,
 type Option func(*Provider)
 
 func WithAddress(address string) Option {
+	if address == "" {
+		address = defaultAddress
+	}
 	return func(p *Provider) {
 		p.address = address
 	}
 }
 
 func WithLogger(logger *zap.Logger) Option {
+	if logger == nil {
+		return func(p *Provider) {}
+	}
```

Comment:
Why do we need those checks?
The default values were already configured [here](https://github.com/pipe-cd/pipe/pull/1694/files#diff-18cf6164f76e899fb1e88f56a41b08fba748b510d0dc5fb6cb76f196159b7882R60).

Reply:
Say the nil given, it will be overridden. I was considering that happens, but you feel too hard?

Reply:
Checking the passing values everywhere will be hard and complicated.
And calling `WithLogger` means the caller wants to customize the logger so I think we don't need to check.

Reply:
Okay. No objection to that!